lib/components/helpers/pretty-text-input.js

```diff
@@ -376,7 +376,12 @@ export default createReactClass({
       });
     };
 
-    this.codeMirror.operation(tagOps);
+    // XXX: Make sure we apply those operation after React has made its rendering pass.
+    // As React 16 is asynchronous and uses rAF, it's same to assume that this will
+    // be called after React has patched the DOM.
+    requestAnimationFrame(() => {
```

Comment:
@jdeal It's kind of hacky solution but it works well.
As we are rendering tags outside the React rendering loop, we want to make sure our tags have been committed to the DOM so CodeMirror can safely do its internal stuff.

Reply:
A cleaner solution would be to get rid of the current system and render everything in the `renderDefault` function switching between the read-only editor and CodeMirror. But I didn't want to mess around with this as it already works.

Reply:
Ooh, nice fix! I was thinking that was going to be nastier. And yeah, I really, really, really would like to switch to Draft.js so we could just render like normal.